atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimestampBoundStoreTest.java

```diff
@@ -0,0 +1,150 @@
+        insertTimestampWithCorrectId(OFFSET);
+        assertBoundInDbIsEqualTo(OFFSET);
+    }
+
+    @Test
+    public void canGetOldFormat() {
+        insertTimestampOld(OFFSET);
+        assertBoundInDbIsEqualTo(OFFSET);
+    }
+
+    @Test
+    public void storeWithWrongTimestampCorrectIdSucceeds() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithCorrectId(limit + OFFSET);
+        store.storeUpperLimit(limit + SECOND_OFFSET);
+        assertBoundInDbIsEqualTo(limit + SECOND_OFFSET);
+    }
+
+    @Test
+    public void storeWithWrongTimestampNoIdThrows() {
+        long limit = store.getUpperLimit();
+        insertTimestampOld(limit + OFFSET);
+        assertThatStoreUpperLimitThrows(limit + SECOND_OFFSET);
+    }
+
+    @Test
+    public void storeWithWrongTimestampWrongIdThrows() {
+        long limit = store.getUpperLimit();
+        insertTimestampWithFakeId(limit + OFFSET);
+        assertThatStoreUpperLimitThrows(limit + SECOND_OFFSET);
+    }
+
+    @Test
+    public void storeWithRightTimestampNoIdSucceeds() {
+        insertTimestampOld(OFFSET);
+        assertThat(store.getUpperLimit()).isEqualTo(OFFSET);
+        store.storeUpperLimit(SECOND_OFFSET);
+        assertBoundInDbIsEqualTo(SECOND_OFFSET);
+    }
+
+    @Test
+    public void storeWithRightTimestampWrongIdSucceeds() {
+        insertTimestampWithFakeId(OFFSET);
+        store.getUpperLimit();
+        store.storeUpperLimit(SECOND_OFFSET);
+        assertBoundInDbIsEqualTo(SECOND_OFFSET);
+    }
+
+    @After
+    public void cleanUp() {
+        kv.dropTable(AtlasDbConstants.TIMESTAMP_TABLE);
+    }
+
+    private void assertBoundInDbIsEqualTo(long limit) {
+        assertThat(store.getUpperLimit()).isEqualTo(limit);
+    }
+
+    private void assertThatStoreUpperLimitThrows(long limit) {
+        assertThatThrownBy(() -> store.storeUpperLimit(limit))
+                .isExactlyInstanceOf(MultipleRunningTimestampServiceError.class);
+    }
+
+    private void insertTimestampWithCorrectId(long value) {
+        insertTimestampWithIdChanged(value, false);
+    }
+
+    private void insertTimestampWithFakeId(long value) {
+        insertTimestampWithIdChanged(value, true);
+    }
+
+    private void insertTimestampOld(long value) {
+        setTimestampTableValueTo(PtBytes.toBytes(value));
+    }
+
+    private void insertTimestampWithIdChanged(long value, boolean changeId) {
+        long id = ((CassandraTimestampBoundStore) store).getId();
+        if (changeId) {
+            id = id % 2 + 1;
+        }
+        setTimestampTableValueTo(PtBytes.toBytes(Long.toString(id) + "_" + Long.toString(value)));
+    }
+
+    private void setTimestampTableValueTo(byte[] data) {
+        kv.truncateTable(AtlasDbConstants.TIMESTAMP_TABLE);
+        kv.put(AtlasDbConstants.TIMESTAMP_TABLE, ImmutableMap.of(Cell.create(PtBytes.toBytes(ROW_AND_COLUMN_NAME),
+                PtBytes.toBytes(ROW_AND_COLUMN_NAME)), data), CASSANDRA_TIMESTAMP);
```

Comment:
nit: alright. Let's try to format this a bit more neatly (as far as the cell creation goes).